pyresample/utils/array.py

```diff
@@ -0,0 +1,150 @@
+    if not isinstance(arr, da.Array):
+        raise TypeError("_dask_conversion expects a dask.Array as input.")
+    if not np.isin(to.lower(), np.char.lower(VALID_FORMAT)):
+        raise ValueError("Valid _dask_conversion array formats are {}".format(VALID_FORMAT))
+    if to.lower() == 'numpy':
+        dst_arr = arr.compute()
+    elif to.lower() == 'dask':
+        dst_arr = arr
+    elif to.lower() == 'dataarray_numpy':
+        dst_arr = xr.DataArray(arr.compute(), dims=dims)
+    else:  # to.lower() == 'dataarray_dask':
+        dst_arr = xr.DataArray(arr, dims=dims)
+    return dst_arr
+
+
+def _xr_numpy_conversion(arr, to):
+    if not isinstance(to, str):
+        raise TypeError("'to' must be a string indicating the conversion array format.")
+    if not isinstance(arr, xr.DataArray):
+        raise TypeError("_xr_numpy_conversion expects a xr.DataArray with numpy array as input.")
+        if not isinstance(arr.data, np.ndarray):
+            raise TypeError("_xr_numpy_conversion expects a xr.DataArray with numpy array as input.")
+    if not np.isin(to.lower(), np.char.lower(VALID_FORMAT)):
+        raise ValueError("Valid _xr_numpy_conversion array formats are {}".format(VALID_FORMAT))
+    if to.lower() == 'numpy':
+        dst_arr = arr.data
+    elif to.lower() == 'dask':
+        dst_arr = da.from_array(arr.data)
+    elif to.lower() == 'dataarray_numpy':
+        dst_arr = arr
+    else:  # to.lower() == 'dataarray_dask':
+        dst_arr = xr.DataArray(da.from_array(arr.data), dims=arr.dims)
+    return dst_arr
+
+
+def _xr_dask_conversion(arr, to):
+    if not isinstance(to, str):
+        raise TypeError("'to' must be a string indicating the conversion array format.")
+    if not isinstance(arr, xr.DataArray):
+        raise TypeError("_xr_dask_conversion expects a xr.DataArray with dask.Array as input.")
+        if not isinstance(arr.data, da.Array):
+            raise TypeError("_xr_dask_conversion expects a xr.DataArray with dask.Array as input.")
+    if not np.isin(to.lower(), np.char.lower(VALID_FORMAT)):
+        raise ValueError("Valid _xr_dask_conversion array formats are {}".format(VALID_FORMAT))
+    if to.lower() == 'numpy':
+        dst_arr = arr.data.compute()
+    elif to.lower() == 'dask':
+        dst_arr = arr.data
+    elif to.lower() == 'dataarray_numpy':
+        dst_arr = arr.compute()
+    else:  # to.lower() == 'dataarray_dask':
+        dst_arr = arr
+    return dst_arr
+
+
+def _convert_2D_array(arr, to, dims=None):
+    """
+    Convert a 2D array to a specific format.
+
+    Useful to return swath lons, lats in the same original format after processing.
+
+    Parameters
+    ----------
+    arr : (np.ndarray, da.Array, xr.DataArray)
+        The 2D array to be converted to another array format.
+    to : TYPE
+        The desired array output format.
+        Accepted formats are: ['Numpy','Dask', 'DataArray_Numpy','DataArray_Dask']
```

Comment:
To add to this and clarify a little, you could use actually class objects for this (`np.ndarray`, `da.Array`, `xr.DataArray`, etc). Or even better (at least that's my current feeling) do the split functions like I mentioned and now that I realize you return the input type, you could instead return the function needed to reverse the conversion.

Or...you could do something like a decorator on the methods that replaces your use of these functions so that the internal implementation of the methods only deals with the object types they want. All the conversion complexity is kept outside of the internals/implementation. I do realize that there are some edge cases where a conversion would be done when nothing was really modified in the data, but if done correctly that shouldn't be too big of a deal.

These are just ideas. I can definitely be convinced otherwise, but the new string names for these things feels like the wrong approach.

Reply:
OK I will take some time to think of your suggestions and try implementing it. 
Just to be sure @djhoese: 
- You mean to pass in the function argument the class object (`np.ndarray, da.Array, xr.DataArray`) we want instead of a string-acronym right? 
- Should I add a `chunked/lazy=True/False` argument to decide whether to return an xr.DataArray with dask or numpy array? Or default to a xr.DataArray with numpy?